src/constraint/unique_constraint.c

```diff
@@ -138,7 +142,9 @@ static bool Constraint_EnforceUniqueEntity
 	if(iter != NULL) {
 		RediSearch_ResultsIteratorFree(iter);
 	} else {
-		RediSearch_QueryNodeFree(root);
+		for(uint8_t i = 0; i < _c->n_attr; i++) {
+			RediSearch_QueryNodeFree(nodes[i]);
+		}
```

Comment:
I was under the impression that freeing `root` will free all of its children recursively